Add  io.github.martinrotter.textosaurus

martinrotter opened the pull request:
https://github.com/martinrotter/textosaurus

TingPing commented:
The app-id should be `io.github.martinrotter.textosaurus`

TingPing commented on com.github.textosaurus.json:
```diff
@@ -0,0 +1,33 @@
+            "buildsystem": "qmake",
+            "post-install": [
+                "install -d /app/share/icons/hicolor/512x512/apps",
+                "install -m644 /app/share/pixmaps/textosaurus.png /app/share/icons/hicolor/512x512/apps/textosaurus.png"
```
Pass `-D` here and you can remove the previous line. You will also want to delete the file from `pixmaps` and honestly just change that upstream as nobody should be using that directory.

TingPing commented:
Review of the appdata file:

- `launchable` refers to wrong desktop file
- missing `screenshots`
- missing `releases`

TingPing commented on com.github.textosaurus.json:
```diff
@@ -0,0 +1,33 @@
+                     "--share=network",
+                     "--socket=x11",
+                     "--socket=wayland",
+                     "--filesystem=xdg-download",
```
Surely as a text editor you want more access than that.

martinrotter replied:
Sure.

martinrotter commented:
> The app-id should be io.github.martinrotter.textosaurus

Why? If I take a look into flathub org github, I see many `com.github.xxxx` apps.

TingPing commented on com.github.textosaurus.json:
```diff
@@ -0,0 +1,46 @@
+{
+    "build-options": {
```
This is a very strange layout. Please be consistent with the order other manifest use.

martinrotter replied:
Is order somehow defined in Flatpak docs? I can reorder it I guess, just autoformatted.

TingPing replied:
No it isn't defined. Alphabetical just doesn't make sense.

TingPing commented on com.github.textosaurus.json:
```diff
@@ -0,0 +1,46 @@
+                "make -j3",
+                "make install"
+            ],
+            "buildsystem": "simple",
```
As I said before, `simple` isn't needed. Use `qmake` and set `config-opts`.

martinrotter replied:
Is there somewhere docs for qmake buildsystem?

TingPing replied:
Not specifically. Here is an example:

```json
  {
    "buildsystem": "qmake",
    "config-opts": [
      "CONFIG+=release"
    ]
   }
```

martinrotter commented:
>All of them contain, or should, the org name. Yours does not. io vs com is less consistent but I think it makes sense to reserve the former for user repos and the latter for official Github projects.
So should I add "io" or stay with "com" and add org name?

TingPing commented:
I think `io.github.martinrotter.textosaurus` is the best choice.

martinrotter commented:
Will do.

martinrotter commented:
@TingPing Kindle requesting review, fixed all suggested changes.

martinrotter commented:
+ fixed paths for flatpak-specific version of Textosaurus according to http://docs.flatpak.org/en/latest/conventions.html#xdg-base-directories, moved icon to hicolor/....

TingPing commented:
bot, build io.github.martinrotter.textosaurus

TingPing commented on io.github.martinrotter.textosaurus.json:
```diff
@@ -0,0 +1,44 @@
+                "QMAKE_CXX=/usr/lib/sdk/gcc7/bin/g++",
+                "CONFIG+=release",
+                "CONFIG+=FLATPAK_MODE",
+                "PREFIX=/app"
```
Prefix is already set for you.

TingPing commented:
> fixed paths for flatpak-specific version of Textosaurus according to http://docs.flatpak.org/en/latest/conventions.html#xdg-base-directories

That looks like a misunderstanding. You shouldn't change anything and the previous behavior looked fine.

martinrotter commented:
"config" folder for application when run under Flatpak is "sandboxed" for the one single app (if I get it correctly), now, the source code **literally** follows Flatpak docs, which says `~/.var/app/<app-id>/config` so I coded/fixed the app to use that path literally = `~/.var/app/io.github.martinrotter.textosaurus/config/config.ini`

TingPing commented:
> @TingPing Check the screenshot, it is taken from app when running from flatpak-builder.

Yes it is wrong and the other codepath was right.

martinrotter commented:
@TingPing Then your docs say it just wrong. But I can revert it, just to prove that I am kind.

martinrotter commented:
@TingPing Fixed it, same textosaurus tag/release, also added OARS to appdata.

TingPing commented:
Anyway everything built fine just update it to the latest commit and ensure you are happy with the results using it.

martinrotter commented:
@TingPing  Wait, what do you mean with " update it to the latest commit"?

TingPing commented:
Oh have you been updating the `2018.04.16` tag? If so please don't do that and also include 
`"commit": "15dcb218150efe2e21de6acaa9614c48c18fa860"` so it is clear what is building.

martinrotter commented:
Yes, I updated the tag.

martinrotter commented:
@TingPing  done

TingPing commented:
bot, build io.github.martinrotter.textosaurus

martinrotter commented:
Thanks. :)

nedrichards commented:
Repository has been created: https://github.com/flathub/io.github.martinrotter.textosaurus
